### ISSUE TITLE
rustdoc: brighten/darken link colors based on background color

### ISSUE THREAD

**Issue by jsha** [rust-lang/rust#93230]
The readability of text depends in part on its contrast relative to the background. For instance, in the light theme:

![image](https://user-images.githubusercontent.com/220205/150669465-e0b56194-fbe6-4c70-8bfd-8d10eb3d18c7.png)

![image](https://user-images.githubusercontent.com/220205/150669467-2ffafbe6-934f-447e-b195-da1d82754e48.png)

![image](https://user-images.githubusercontent.com/220205/150669471-d8113920-7b9e-42f6-ba0c-f16e720bc275.png)

You can see that the contrast ratio is different depending on the background. Against a white background it's 4.97. Against a grey background it's 4.56. Not a huge difference, but noticeable. For reference, the WCAG guideline is that [text should have a contrast ratio of at least 4.56](https://webaim.org/articles/contrast/#sc143).

We should pick an alternate link color that maintains the same contrast ratio when placed against the grey backgrounds we use (code highlighting and sidebar).

The same concept applies to the dark theme, in reverse - the link color should be brightened when it is against a lighter background, so it stays equally visible.

**Comment by jsha** [comment]
Fixed in #93231
